modules/fedimint-ln-client/src/lib.rs

```diff
@@ -271,10 +271,32 @@ impl LightningClientExt for Client {
             return Ok(completed_payment);
         }
 
+        // Verify that no previous payment attempt is still running
+        let prev_operation_id = lightning
+            .get_payment_operation_id(invoice.payment_hash(), prev_payment_result.index)
+            .await;
+        if self.has_active_states(prev_operation_id).await {
+            return Err(anyhow::anyhow!("Previous payment attempt still in progress. Previous Operation Id: {prev_operation_id}"));
+        }
```

Comment:
I think this works now if we don't have any bugs in the state machines. As a second line of defense we could also check that the contract is empty after making sure no SMs are running.

Reply:
I added a check below. If the contract exists, it must have a value of 0, otherwise this function will return an error.